providers.d/apt-mirror/provider_wrapper.sh

```diff
@@ -1,13 +1,37 @@
 bashelliteProviderWrapper() {
 
-  if [[ ! ${_r_dryrun} ]]; then
-    apt-mirror ${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf;
+  local basepath_parameter="$(grep -oP "(?<=(^set base_path[[:space:]]*))[[:alnum:]/]*" ${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf)"
```

Comment:
Forward look ups do not allow for wildcard characters. A viable alternative might be something like:

```grep -oP "(?<=(^set[[:blank:]]base_path[[:blank:]])).*"```

It really doesn't matter what is behind base_path, all of it should be compared to the mirror_tld + repo_name.

Reply:
The problem here is that the value can be any number greater than 1 whitespace from "set base_path".  So we can't just grab what is after "set base_path" for comparison.  We need a way to strip any whitespace before the value.  I tried this and it works.  But I don't know if there is better way....

grep -oP "(?<=^set base_path).*" test.txt | grep -oP "/[[:alnum:]/]+"

Reply:
As discussed, here is the error I receive when trying your original method: 

```
root@chrx:~# cat test.txt 
set    base_path    some_base_path
root@chrx:~# grep -oP "(?<=(^set base_path[[:space:]]*))[[:alnum:]/]*" test.txt
grep: lookbehind assertion is not fixed length
root@chrx:~# grep --version | head -n 1
grep (GNU grep) 2.25
```
You can use something like this... it is more lines, but only one grep call:

```
root@chrx:~# cat test.txt 
set    base_path    some_base_path
root@chrx:~# base_path=$(egrep "^\s*set\s*base_path\s*[[:alnum:]/_-]*" test.txt);
root@chrx:~# echo $base_path
set base_path some_base_path
root@chrx:~# base_path="${base_path#*set*base_path}"
root@chrx:~# echo $base_path
some_base_path
```

By not quoting the output of the egrep, it drops the leading spaces automagically. You just have to be cautious with your regex to ensure you are validating/sanitizing input from the egrep.

Reply:
Updated.